services/src/main/java/org/keycloak/services/resources/admin/GroupsResource.java

```diff
@@ -124,11 +121,14 @@ public Stream<GroupRepresentation> getGroups(@QueryParam("search") String search
     @Path("{group-id}")
     public GroupResource getGroupById(@PathParam("group-id") String id) {
         GroupModel group = realm.getGroupById(id);
+
         if (group == null) {
             throw new NotFoundException("Could not find group by id");
         }
 
-        checkForOrgRelatedGroupModel(session, group);
+        if (!Organizations.canManageOrganizationGroup(session, group)) {
```

Comment:
We can safely remove the validation from here and have the organization provider as the source of truth for validations, so a single place where it happens.

From an UI perspective, we could probably show organizational groups disabled based on the `kc.org` attribute.